tests/containers/image.pm

```diff
@@ -27,14 +27,18 @@ sub run {
     # We may test either one specific image VERSION or comma-separated CONTAINER_IMAGE_VERSIONS
     my $versions = get_var('CONTAINER_IMAGE_VERSIONS', get_required_var('VERSION'));
     for my $version (split(/,/, $versions)) {
-        my ($untested_images, $released_images) = get_suse_container_urls(version => $version);
-        my $images_to_test = check_var('CONTAINERS_UNTESTED_IMAGES', '1') ? $untested_images : $released_images;
+        my $images_to_test;
+        unless ($images_to_test = get_urls_from_var()) {
```

Comment:
This block is very difficult to understand. Please add a comment that explains, that this checks for the list to be empty.

```suggestion
        unless ($images_to_test = get_urls_from_var("CONTAINER_IMAGES_TO_TEST")) {
            # If the returned list is empty, get the default images
```

Even better for the understanding would be to check if the variable is present, e.g.

```suggestion
        if (get_var('CONTAINERS_UNTESTED_IMAGES', '') ne '') {
            $images_to_test = get_urls_from_var();
        } else {
             my ($untested_images, $released_images) = get_suse_container_urls(version => $version);
             $images_to_test = check_var('CONTAINERS_UNTESTED_IMAGES', '1') ? $untested_images : $released_images;
        }
```

Reply:
I added the comment as I don't understand the code you suggested. Please check again.